src/XMakeTasks/UnitTests/AssemblyDependency/AssemblyFoldersFromConfig_Tests.cs

```diff
@@ -0,0 +1,150 @@
+        public void AssemblyFoldersFromConfigTest()
+        {
+            var assemblyConfig = Path.GetTempFileName();
+            File.WriteAllText(assemblyConfig, TestFile);
+
+            var moniker = "{AssemblyFoldersFromConfig:" + assemblyConfig + ",v4.5}";
+
+            try
+            {
+                ResolveAssemblyReference t = new ResolveAssemblyReference
+                {
+                    BuildEngine = new MockEngine(),
+                    Assemblies = new ITaskItem[] {new TaskItem("assemblyfromconfig2")},
+                    SearchPaths = new[] {moniker}
+                };
+                
+                Execute(t);
+
+                Assert.Equal(1, t.ResolvedFiles.Length);
+                Assert.Equal(@"c:\assemblyfromconfig\folder2\assemblyfromconfig2.dll", t.ResolvedFiles[0].ItemSpec);
+                AssertNoCase(moniker, t.ResolvedFiles[0].GetMetadata("ResolvedFrom"));
+            }
+            finally
+            {
+                FileUtilities.DeleteNoThrow(assemblyConfig);
+            }
+        }
+
+        [Fact]
+        public void AssemblyFoldersFromConfigPlatformSpecificAssemblyFirstTest()
+        {
+            var assemblyConfig = Path.GetTempFileName();
+            File.WriteAllText(assemblyConfig, TestFile);
+
+            var moniker = "{AssemblyFoldersFromConfig:" + assemblyConfig + ",v4.5}";
+
+            try
+            {
+                ResolveAssemblyReference t = new ResolveAssemblyReference();
+
+                t.BuildEngine = new MockEngine();
+
+                t.Assemblies = new ITaskItem[] { new TaskItem("assemblyfromconfig_common.dll") };
+                t.SearchPaths = new[] { moniker };
+                t.TargetProcessorArchitecture = "x86";
+                Execute(t);
+
+                Assert.Equal(1, t.ResolvedFiles.Length);
+                Assert.Equal(@"c:\assemblyfromconfig\folder_x86\assemblyfromconfig_common.dll", t.ResolvedFiles[0].ItemSpec);
+                AssertNoCase(moniker, t.ResolvedFiles[0].GetMetadata("ResolvedFrom"));
+            }
+            finally
+            {
+                FileUtilities.DeleteNoThrow(assemblyConfig);
+            }
+        }
+
+        [Fact]
+        public void AssemblyFoldersFromConfigFileNotFoundTest()
+        {
+            var assemblyConfig = Path.GetTempFileName();
+            File.Delete(assemblyConfig);
+            var moniker = "{AssemblyFoldersFromConfig:" + assemblyConfig + ",v4.5}";
+
+            try
+            {
+                ResolveAssemblyReference t = new ResolveAssemblyReference();
+
+                t.BuildEngine = new MockEngine();
+
+                t.Assemblies = new ITaskItem[] { new TaskItem("assemblyfromconfig_common.dll") };
+                t.SearchPaths = new[] { moniker };
+                t.TargetProcessorArchitecture = "x86";
+                var success = Execute(t);
+                Assert.True(success);
+
+                Assert.Equal(0, t.ResolvedFiles.Length);
```

Comment:
I assume this is mapped from the other tests--test for success and 0 length returns, no test for warnings?

Are you not failing on explicitly-passed-in-nonexistent-file because you've changed common targets to pass in the path unconditionally? That seems ok, but worth being explicit.